### PR TITLE
Improve ctkSliderWidget and ctkVTKDiscretizableColorTransferWidget

### PR DESCRIPTION
**Pull request by tcoulange**
- Add support for custom spinbox limits in ctkSliderWidget
- Improve ctkVTKDiscretizableColorTransferWidget:
  - Supports any spinbox value
  - Split setHistogramConnection and updateHistogram
  - Replace std::cout by qDebug
  - Fix some bugs

### CODE REVIEW

File: Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp [hunk 1]
```diff
@@ -33,6 +33,9 @@
 #include <vtkTable.h>
 #include <vtkTransform2D.h>
 
+// STD includes
+#include <qdebug.h>
```
Comment (jcfr) [inline Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp]:
```
#include <QDebug>
```

File: Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp [hunk 2]
```diff
@@ -33,6 +33,9 @@
 #include <vtkTable.h>
 #include <vtkTransform2D.h>
 
+// STD includes
```
Comment (jcfr) [inline Libs/Visualization/VTK/Core/vtkDiscretizableColorTransferChart.cpp]:
```
// Qt includes
```

File: Libs/Widgets/ctkRangeWidget.cpp [hunk 3]
```diff
@@ -211,6 +217,26 @@ void ctkRangeWidgetPrivate::relayout()
...
 }
 
+// --------------------------------------------------------------------------
+bool ctkRangeWidgetPrivate::useCustomSpinBoxesLimits()const
+{
+  return this->CustomSpinBoxesLimitsMin <= this->CustomSpinBoxesLimitsMax;
+}
+
+// --------------------------------------------------------------------------
+double ctkRangeWidgetPrivate::clamp(double v, double min, double max)
```
Comment (jcfr) [inline Libs/Widgets/ctkRangeWidget.cpp]: Consider using `qBound`

**Comment by jcfr** [comment]
Should any of the test be updated ?

File: Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp [hunk 4]
```diff
@@ -387,20 +389,39 @@ const
...
 {
   Q_D(ctkVTKDiscretizableColorTransferWidget);
 
-  this->initializeHistogramAndDataRange(input);
-  if (useInputDataRange)
+  if (!input)
   {
-    // update visible range, which updates histogram
-    this->resetVisibleRange(ResetVisibleRange::UNION_DATA_AND_CTF);
+    d->histogramFilter = nullptr;
```
Comment (jcfr) [inline Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp]: `nullptr` -> `CTK_NULLPTR`

**Comment by jcfr** [comment]
Closing. Integrated in https://github.com/commontk/CTK/commit/430d934e605b3401460af1d2a78f78c000c4c3b5

**Comment by tcoulange** [comment]
Thank you for the review. It seems like you fixed most of the issues, I will fix the last ones in a new pull request.

**Comment by jcfr** [comment]
No problem. Thanks for your contributions :)

If possible, do not hesitate to add or update tests (even if minimal)

**Comment by tcoulange** [comment]
Yes sure, I will improve the tests.

**Pull request closed without merging**